diff --git a/sunnypilot/models/manager.py b/sunnypilot/models/manager.py
index 891313520..a3ce046a3 100644
--- a/sunnypilot/models/manager.py
+++ b/sunnypilot/models/manager.py
@@ -8,7 +8,6 @@ See the LICENSE.md file in the root directory for more details.
 import asyncio
 import os
 import time
-import json
 
 import aiohttp
 from openpilot.common.params import Params
@@ -146,7 +145,7 @@ class ModelManagerSP:
       await asyncio.gather(*tasks)
       self.active_bundle = self.selected_bundle
       self.active_bundle.status = custom.ModelManagerSP.DownloadStatus.downloaded
-      self.params.put("ModelManager_ActiveBundle", json.dumps(self.active_bundle.to_dict()))
+      self.params.put("ModelManager_ActiveBundle", self.active_bundle.to_dict())
       self.selected_bundle = None
 
     except Exception:
@@ -169,16 +168,16 @@ class ModelManagerSP:
         self.available_models = self.model_fetcher.get_available_bundles()
         self.active_bundle = get_active_bundle(self.params)
 
-        if index_to_download := self.params.get("ModelManager_DownloadIndex", block=False, encoding="utf-8"):
-          if model_to_download := next((model for model in self.available_models if model.index == int(index_to_download)), None):
+        if index_to_download := self.params.get("ModelManager_DownloadIndex"):
+          if model_to_download := next((model for model in self.available_models if model.index == index_to_download), None):
             try:
               self.download(model_to_download, Paths.model_root())
             except Exception as e:
               cloudlog.exception(e)
             finally:
-              self.params.put("ModelManager_DownloadIndex", "")
+              self.params.remove("ModelManager_DownloadIndex")
 
-        if self.params.get("ModelManager_ClearCache", block=False, encoding="utf-8"):
+        if self.params.get("ModelManager_ClearCache"):
             self.clear_model_cache()
             self.params.remove("ModelManager_ClearCache")
 
